File: MRooms/Finals/AndroidCheckBox/app/src/main/java/com/glen/androidcheckbox/MainActivity.java
package com.glen.androidcheckbox;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    CheckBox pizza,coffee,burger;
    Button buttonOrder;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        addListenerOnButtonClick();
    }
    public void addListenerOnButtonClick(){
        //Getting instance of CheckBoxes and Button from the activty_main.xml file
        pizza=(CheckBox)findViewById(R.id.checkBox);
        coffee=(CheckBox)findViewById(R.id.checkBox2);
        burger=(CheckBox)findViewById(R.id.checkBox3);
        buttonOrder=(Button)findViewById(R.id.button);

        //Applying the Listener on the Button click
        buttonOrder.setOnClickListener(new View.OnClickListener(){

            @Override
            public void onClick(View view) {
                int totalamount=0;
                StringBuilder result=new StringBuilder();
                result.append("Selected Items: ");
                if(pizza.isChecked()){
                    result.append("Pizza 100Rs ");
                    totalamount+=100;
                }
                if(coffee.isChecked()){
                    result.append("Coffee 50Rs ");
                    totalamount+=50;
                }
                if(burger.isChecked()){
                    result.append("Burger 120Rs ");
                    totalamount+=120;
                }
                result.append("Total: "+totalamount+"Rs");
                //Displaying the message on the toast
                Toast.makeText(getApplicationContext(), result.toString(), Toast.LENGTH_LONG).show();
            }

        });
    }
}

File: MidtermExam2/app/src/main/java/com/glen/midtermexam2/MainActivity.kt
package com.glen.midtermexam2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    private lateinit var todoAdapter: TodoAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        todoAdapter = TodoAdapter(mutableListOf())

        val rvTodo = findViewById<RecyclerView>(R.id.rv_todo)
        rvTodo.adapter = todoAdapter

        rvTodo.layoutManager = LinearLayoutManager(this)

        val btAdd = findViewById<Button>(R.id.bt_add)
        btAdd.setOnClickListener {
            val todoTitle = findViewById<EditText>(R.id.et_todo).text.toString()
            if(todoTitle.isNotEmpty()) {
                val todo = Todo(todoTitle)
                todoAdapter.addTodo(todo)
                findViewById<EditText>(R.id.et_todo).text.clear()
            }
        }

        val btClear = findViewById<Button>(R.id.bt_clear)
        btClear.setOnClickListener {
            todoAdapter.deleteDoneTodos()
        }
    }

    //show main_menu.xml
    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_game -> {
                Toast.makeText(this, "Game", Toast.LENGTH_SHORT).show()
                return true
            }
            R.id.action_shop -> {
                Toast.makeText(this, "Shop", Toast.LENGTH_SHORT).show()
                return true
            }
            R.id.action_stats -> {
                Toast.makeText(this, "Stats", Toast.LENGTH_SHORT).show()
                return true
            }
            R.id.action_mode -> {
                Toast.makeText(this, "Mode", Toast.LENGTH_SHORT).show()
                return true
            }
            R.id.action_about -> {
                Toast.makeText(this, "About", Toast.LENGTH_SHORT).show()
                return true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}